Examples/JavaScript/JavaScript Essentials/Functions.ts

```diff
@@ -0,0 +1,63 @@
+//// { order: 2 }
+
+// There are quite a few ways to declare a function in JavaScript, let's look at
+// a function which adds two numbers together:
+
+// @ts-ignore
+// Creates a function in global scope called addOldSchool
+function addOldSchool(x, y) { return x + y }
+
+// @ts-ignore
+// You can move the name of the function to a variable name also
+const anonymousOldSchoolFunction = function(x, y) { return x + y }
+
+// @ts-ignore
+// You can also use fat-arrow shorthand for a function
+const addFunction = (x, y) => { return x + y }
+
+// We're going to focus on the last one, but everything applies to all three formats.
+
+// TypeScript provides additional syntax which wraps functions to provide hints on what
+// the types are expected by this function. This is the most open version of the function, 
+// it says that add takes two inputs of any type: this could be strings, numbers or 
+// objects which you've made.
+
+const add1 = (x: any, y: any) => { return x + y }
+add1("Hello", 23)
+
+// This is legitimate JavaScript (strings can be added like this for example) but isn't optimal
+// for our function which we know is for numbers, so we'll convert the x and y to only be numbers
+
+const add2 = (x: number, y: number) => { return x + y }
+add2(16, 23)
+add2("Hello", 23)
+//   ~~~~~~~ Argument of type '"Hello"' is not assignable to parameter of type 'number'.
```

Comment:
I think I would remove this comment and change the next paragraph to say something like "Great. You can see from the squiggly underline that we get an error when ..."